devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/tooling/dependency/DependencyUtils.java

```diff
@@ -378,8 +377,7 @@ private static Dependency createDeploymentProjectDependency(DependencyHandler ha
                     ped.getDeploymentModule().getName(),
                     ped.getDeploymentModule().getVersion().toString());
         } else if (ped.getDeploymentModule() instanceof ProjectInternal) {
-            return handler.create(new DefaultProjectDependency((ProjectInternal) ped.getDeploymentModule(), true,
-                    DefaultTaskDependencyFactory.withNoAssociatedProject()));
+            return handler.create(new DefaultProjectDependency(((ProjectInternal) ped.getDeploymentModule()).getOwner()));
```

Comment:
So there's no way to make this work for both Gradle 8.x and 9. If so, we might have to use reflection.

/cc @cdsap